tests/auto/tst_persistenttabmodel/tst_persistenttabmodel.cpp

```diff
@@ -428,11 +494,45 @@ void tst_persistenttabmodel::onTitleChanged()
 
     QSignalSpy dataChangedSpy(tabModel, SIGNAL(dataChanged(QModelIndex, QModelIndex, QVector<int>)));
 
-    mockPage.m_tabId = 1;
-    mockPage.setTitle("Hello world");
+    EXPECT_CALL(mockPage, tabId()).WillOnce(Return(1));
+    QUrl url;
+    EXPECT_CALL(mockPage, url()).WillOnce(Return(url));
+    EXPECT_CALL(mockPage, title()).WillOnce(Return(QString("Hello world")));
+    emit mockPage.titleChanged();
+
     QCOMPARE(dataChangedSpy.count(), 1);
 }
 
+void tst_persistenttabmodel::nextActiveTabIndex()
+{
+    DeclarativeWebContainer container;
+    DeclarativeWebPage page;
+    tabModel->setWebContainer(&container);
+
+    EXPECT_CALL(container, webPage()).WillRepeatedly(Return(&page));
+    EXPECT_CALL(page, parentId()).WillOnce(Return(1));
+    EXPECT_CALL(page, tabId()).WillOnce(Return(2));
+    tabModel->nextActiveTabIndex(0);
+}
+
+void tst_persistenttabmodel::misc()
+{
+    // Call simple methods to increase coverage
+
+    tabModel->roleNames();
+    QCOMPARE(tabModel->waitingForNewTab(), false);
+
+    QSignalSpy loadedChangedSpy(tabModel, SIGNAL(loadedChanged()));
+    tabModel->m_loaded = true;
+    tabModel->setUnloaded();
+    QCOMPARE(loadedChangedSpy.count(), 1);
```

Comment:
I'd prefer to have test cases corresponding to tested methods and not to mix them.

Any method is just a function which produces some output for some given input. In C++ the input also includes object state. Any usage scenario can be described in the terms of input and output, hence it should be enough to provide sufficient input in the *_data() methods in order to test any possible scenario. This approach also implies that overloaded methods should be tested separately because they accept different input (activateTab(int), activateTab(QString)).

There might be situations when a tested method calls some other object (usually mocked one) which calls back synchronously and modifies the state of the tested object, but this smells like a bad architecture.